molecule_libvirt/playbooks/create.yml

```diff
@@ -0,0 +1,259 @@
+        {{ disk_size | default(default_disk_size ) }}"
+      args:
+        chdir: "{{ libvirt_default_image_path }}"
+      environment:
+        LIBGUESTFS_BACKEND: direct
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Expand original image on the new sized volume
+      command:
+        "virt-resize --quiet --expand /dev/sda1 {{ item.name }}.qcow2
+        {{ item.name }}-temp.qcow2"
+      args:
+        chdir: "{{ libvirt_default_image_path }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Move temporary file to main image
+      command:
+        "mv {{ item.name }}-temp.qcow2 {{ item.name }}.qcow2"
+      args:
+        chdir: "{{ libvirt_default_image_path }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host if item.libvirt_host is defined else 'localhost' }}"
+
+    - name: Define vm from xml file
+      virt:
+        command: define
+        xml: "{{ lookup('template', 'vm/vm.xml.j2') }}"
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+
+    - name: Start molecule vm(s)
+      virt:
+        name: "{{ item.name }}"
+        command: start
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      register: server
+
+    - name: Get instance(s) ip address
+      shell: >
+        virsh --connect {{ qemu_uri }}
+        domifaddr {{ item.name }} |
+        grep {{ mac }} |
+        awk '{ print $4 }' |
+        sed 's/\/.*//'
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      register: instance_ip_address_local
+      until: instance_ip_address_local.stdout != ""
+      retries: 10
+      delay: 2
+      vars:
+        qemu_uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      when: item.bridge_name is not defined
+
+    - name: Wait 35 seconds for arp table update
+      pause:
+        seconds: 35
+      vars:
+        jmesquery: "[].bridge_name"
+      when: (molecule_yml.platforms | from_yaml | json_query(jmesquery)) != []
+
+    - name: Get instance(s) running on remote host(s)
+      shell: >
+        arp -n {{ item.name }} |
+        grep -oE "\b([0-9]{1,3}\.){3}[0-9]{1,3}\b"
+      with_items: "{{ molecule_yml.platforms }}"
```

Comment:
Do we want to keep `with_items` or do we want to use `loop`?

Reply:
`with_items` has been changed for `loop`.